src/redux/mission/missionReducer.js

```diff
@@ -0,0 +1,32 @@
+import MissionActionTypes from './missionType';
+
+const INITIAL_STATE = {
+  loading: true,
+  missions: [],
+  error: false,
+  joined_mission: [],
+};
+
+const MissionReducer = (state = INITIAL_STATE, { payload, type }) => {
+  switch (type) {
+    case MissionActionTypes.FETCH_MISSIONS_START:
+      return { ...state, loading: true };
+    case MissionActionTypes.FETCH_MISSIONS_SUCCESS:
+      return {
+        ...state, loading: false, error: false, missions: payload,
+      };
+    case MissionActionTypes.FETCH_MISSIONS_ERROR:
+      return { ...state, error: payload };
+    case MissionActionTypes.JOIN_MISSION:
+      return { ...state, joined_mission: [...state.joined_mission, payload] };
```

Comment:
[OPTIONAL] This is a nice way to keep track of reserved assets but please stick to activity guidelines next time for the implementation method. Because it would be important to deliver the task as required by the customer.

![reserved](https://user-images.githubusercontent.com/83017311/162006350-551196ba-f919-4466-b9ff-2dd2d1aa75ef.png)